multinet/multinet/multinet.py

```diff
@@ -0,0 +1,286 @@
+"""Flask blueprint for Multinet REST API."""
+import csv
+from graphql import graphql
+from io import StringIO
+import itertools
+import json
+import newick
+import re
+import uuid
+
+from flask import Blueprint, request
+from flask import current_app as app
+
+from .schema import schema
+from . import db
+
+bp = Blueprint('multinet', __name__, url_prefix='/multinet')
+
+
+def graphql_query(query, variables=None):
+    """Perform a GraphQL query using optional variable definitions."""
+    result = graphql(schema, query, variables=variables or {})
+    if result:
+        errors = [error.message for error in result.errors] if result.errors else []
+        app.logger.warn('Errors in request: %s' % len(errors))
+        for error in errors[:10]:
+            app.logger.warn(error)
+    else:
+        errors = []
+    return dict(data=result.data, errors=errors, query=query)
```

Comment:
It seems here that if the else cause was reached here, then result.data would not be defined. I would propose before checking result, doing something like 
```
data = {}
errors = []
```

And then conditionally overwriting them by checking each, that way there's a default value defined. Then instead of `data=result.data`, it would just be `data=data`. Please lmk what you think

Reply:
Thanks, good catch. Check out 9785cf8.